### ISSUE TITLE
[BUG] False positive; return outside scope of Span

### ISSUE THREAD

**Issue by cbandy** [jjti/go-spancheck#25]
**Describe the bug**

When a span is created inside a block (if, for, etc), the record-error check will complain about `return` statements outside that block.

**To Reproduce**

Below, the `span` variable does not exist outside the `if` block. The `return` statement does not occur "inside" this span.

```go
package main

import (
  "context"
  "errors"

  "go.opentelemetry.io/otel"
)

func main() {
  run(context.Background())
}

func run(ctx context.Context) error {
  if true {
    _, span := otel.Tracer("foo").Start(ctx, "bar")
    span.End()
  }

  return errors.New("asdf")
}
```

**Expected behavior**

I expected no results from the record-error check, as no spans are in scope at the `return`.

**Comment by jjti** [comment]
thank you @cbandy for reporting this!

**Comment by jjti** [comment]
@cbandy I think this is fixed now: https://github.com/jjti/go-spancheck/releases/tag/v0.6.3

The change itself, has the test case: https://github.com/jjti/go-spancheck/commit/e234cf6dd57208a861eb4b711cdb456fdef47251

After you've upgraded (which might be in awhile if you're using golangci-lint) please let me know if you run into any more issues: https://github.com/jjti/go-spancheck/commit/e234cf6dd57208a861eb4b711cdb456fdef47251

**Comment by jjti** [comment]
note: can also test whether this fixes it with: https://github.com/jjti/go-spancheck?tab=readme-ov-file#cli
